Makefile

```diff
@@ -22,10 +27,52 @@ init-ostree-repo:
 	ostree --repo=build-repo init --mode=bare-user
 	ostree --repo=repo init --mode=archive
 
+.PHONY: check-tools
+check-tools:
+	which ostree
+	which rpm-ostree
+	which rpmdistro-gitoverlay
+	which imagefactory
+	which qemu-img
+	which mock
+
+
+# Runs all targets needed to create an image
+.PHONY: build-rhcos
+build-rhcos: check-tools clean repo-refresh rdgo rpmostree-compose os-image
+
+# Cleans up
+.PHONY: clean
+clean:
+	rm -rf build build-0 build-1 src snapshot ${REPO}
+
+# Pulls sources, builds packages, and creates a dnf repo for use
+.PHONY: rdgo
+rdgo:
+	rpmdistro-gitoverlay init
+	rpmdistro-gitoverlay resolve --fetch-all
+	rpmdistro-gitoverlay build
+
+
+# Composes an ostree
 .PHONY: rpmostree-compose
 rpmostree-compose: ${ROOT_DIR}/openshift.repo init-ostree-repo
+	mkdir -p ${REPO}
+	ostree init --repo=${REPO} --mode=archive
 	if test -d cache; then cachedir='--cachedir $(shell pwd)/cache'; fi && \
 	  cd ${ROOT_DIR} && set -x && \
 	  rpm-ostree compose tree $(COMPOSEFLAGS) $${cachedir:-} --repo=$(shell pwd)/build-repo host.yaml
 	ostree --repo=repo pull-local build-repo
 	ostree --repo=repo summary -u
+
+# Makes an image using the ostree
+.PHONY: os-image
+os-image:
+	ostree --repo=repo remote add rhcos --no-gpg-verify ${OSTREE_INSTALL_URL}
+	ostree --repo=repo pull --mirror --commit-metadata-only rhcos
```

Comment:
Hmm...I think if you're doing local builds you want to be using the local ostree repo.  That's what I do.  

The annoying pain point is it requires setting up a webserver.  See https://github.com/projectatomic/rpm-ostree-toolbox/pull/68 and there's other PRs/issues around this too.